src/components/ContactsAdd.js

```diff
@@ -1,31 +1,80 @@
-import { useState } from "react"
+import { useEffect, useState } from "react"
 import { useNavigate } from "react-router-dom";
 
 function ContactsAdd(props) {
+ const navigate = useNavigate();
 
   // setContacts and contacts must be passed as props
   // to this component so new contacts can be added to the
   // state
   const { setContacts, contacts } = props
+  const [ firstName, setFirstName] = useState("")
+  const [ lastName, setLastName] = useState("")
+  const [ street, setStreet] = useState("")
+  const [ city, setCity] = useState("")
+
 
   //TODO: Implement controlled form
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.log("handleSubmit called", contacts)
+    const newContact = { firstName, lastName, street, city}
+
+    fetch("http://localhost:4000/contacts", {
+      method: 'POST',
+      headers: {"Content-Type": 'application/json'},
+      body: JSON.stringify(newContact)
+    }).then(() => {
+      console.log("New Contact added")
+      setFirstName("")
+      setLastName("")
+      setStreet("")
+      setCity("")
+    })  
+    // navigate("/*")
+  }
+
+  const handleChange = (event) => {
+    const value = event.target.value
+    const name = event.target.name
+    // console.log(value, name)
+
+    if (name === "firstName") {
+      setFirstName(value)
+    }
+
+    if (name === "lastName") {
+      setLastName(value)
+    }
+
+    if (name === "street") {
+      setStreet(value)
+    }
+
+    if (name === "city") {
+      setCity(value)
+    }
+  }
+
```

Comment:
i did it slightly different but your solution is very nice! I've never seen fetch usage with many states